File: DSA/Java/src/dsa/lib/algo/sort/RadixSort.java
package dsa.lib.algo.sort;

import dsa.lib.algo.sort.countingSort.CountingSort;
import dsa.lib.utils.Range;

class RadixSort {
    static void sort(int[] nums) {
        if (nums.length < 2) return;
        final int max = Range.max(nums);
        for (int place = 1; place <= max; place *= 10) CountingSort.sort(nums, place);
    }
}

File: DSA/Python/src/dsa/lib/algo/sort/radix_sort.py
from typing import List, AnyStr

from dsa.lib.algo.sort.counting_sort import counting_sort
from dsa.lib.utils import lists


def sort(nums: List[int]) -> None:
    if len(nums) < 2: return

    max_ = max(nums)
    place = 1
    while place <= max_:
        counting_sort.sortIntPlace(nums, place)
        place *= 10


def sortStr(ss: List[AnyStr]) -> None:
    if len(ss) < 2: return

    max_ = lists.maxSize(ss)
    i = len(max_) - 1
    while i >= 0:
        counting_sort.sortStr(ss, i)
        i -= 1
